sbt-plugin/src/main/scala-sbt-0.13/org/scalajs/sbtplugin/SBTCompat.scala

```diff
@@ -0,0 +1,63 @@
+package org.scalajs.sbtplugin
+
+import sbt._
+
+import org.scalajs.core.tools.io.FileVirtualFile
+
+private[sbtplugin] object SBTCompat {
+  type IncOptions = sbt.inc.IncOptions
+
+  type CompileAnalysis = sbt.inc.Analysis
+
+  class sbtUnchecked extends scala.annotation.Annotation
+
+  val formatImplicits: sbt.Cache.type = sbt.Cache
+
+  def moduleIDWithConfigurations(moduleID: ModuleID,
+      configurations: Option[String]): ModuleID = {
+    moduleID.copy(configurations = configurations)
+  }
```

Comment:
FTR, sbt 0.13.16-RC1 added a backport of `withConfigurations` to `ModuleID`, so we could get away without this particular shim in `SBTCompat`. However that would definitely force our users to upgrade to sbt 0.13.16-RC1+ as well (and I'm pretty sure that failing to do so results in ugly obscure error messages like `NoSuchMethodError`), so I wanted to avoid that.